src/main.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import ReactDOM from 'react-dom/client'
+import App from './App.jsx'
+import ProductsPage from './components/ProductsPage';
+import ProductDetail from './components/ProductDetail';
+import HomePage from './components/HomePage';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import './index.css'
+
+ReactDOM.createRoot(document.getElementById('root')).render(
+  <React.StrictMode>
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<App />}>
+          <Route index element={<HomePage />} />
+          <Route path="products" element={<ProductsPage />} />
+          <Route path="productDetail" element={<ProductDetail />} >
```

Comment:
I'd recommend using `products/:productId` instead. For one, I can just type a number at the end my url to get to a specific product instead of having to also replace `products` with `productDetail` and it's typically how these endpoints are labeled. More on that here if interested https://restfulapi.net/resource-naming/.